### PR TITLE
add symmetry information to DomainIntegral in Jintegral tests

### PR DESCRIPTION
**Pull request by BoZeng1997**
add symmetry information to DomainIntegral in J integral tests so that the test results match the analytical value. Also updated the gold files. #24748

### CODE REVIEW

File: modules/tensor_mechanics/test/tests/j_integral/gold/j_integral_2d_mouth_dir_out.csv [hunk 1]
```diff
@@ -1,4 +1,3 @@
 time,J_1,J_2,J_3,J_4,J_5
 0,0,0,0,0,0
-1,1.1655970440759,1.1625839930871,1.1613679143305,1.164168518934,1.1632303584855
```
Comment (recuero) [inline modules/tensor_mechanics/test/tests/j_integral/gold/j_integral_2d_mouth_dir_out.csv:3]: Why don't we get exactly double on this file?

Comment (BoZeng1997) [reply modules/tensor_mechanics/test/tests/j_integral/gold/j_integral_2d_mouth_dir_out.csv:3]: I do not have an explanation. From the code in Jintegral.C
https://github.com/idaholab/moose/blob/51557c3cd1fd65ea9518d83a221f514524c39c99/modules/tensor_mechanics/src/vectorpostprocessors/JIntegral.C#L228-L229
the only place where the symmetry information takes effect is here, by doubling the j_integral result.
`modules/tensor_mechanics/test/tests/j_integral/gold/j_integral_3d_mouth_dir_out.csv` this one is the other result that does not give us exact doubling. They both use `crack_direction_method = CrackMouth` and default `crack_end_direction_method=NoSpecialTreatment`. The other tests with `crack_direction_method = CrackMouth` but `crack_end_direction_method=CrackDirectionVector` satisfy the double relationship.

Comment (recuero) [reply modules/tensor_mechanics/test/tests/j_integral/gold/j_integral_2d_mouth_dir_out.csv:3]: There is a piece of code that modifies the location of the crack front nodes here:

https://github.com/idaholab/moose/blob/51557c3cd1fd65ea9518d83a221f514524c39c99/modules/tensor_mechanics/src/userobjects/CrackFrontDefinition.C#L1063-L1066

I don't fully understand why we are zeroing the components of the symmetry plane for the entire crack mouth definition. It seems as if the nodes of the crack mouth could potentially move along the symmetry. If you change the symmetry plane to "2", you get exactly double the value (but it's obviously wrong in this case).

Comment (recuero) [reply modules/tensor_mechanics/test/tests/j_integral/gold/j_integral_2d_mouth_dir_out.csv:3]: @bwspenc: We can merge this PR once the tests pass, but I think we should address the reason why we don't get exactly double when we add the correct symmetry plane. I can open an issue if that cannot be address now.

File: modules/tensor_mechanics/test/tests/j_integral/gold/j_integral_3d_out.e
Comment (recuero) [inline modules/tensor_mechanics/test/tests/j_integral/gold/j_integral_3d_out.e:1]: Why do we need to regold this file?

Comment (BoZeng1997) [reply modules/tensor_mechanics/test/tests/j_integral/gold/j_integral_3d_out.e:1]: the postprocessor values were written in exodus output. when postprocessor results are updated, all exodus results need to be regold.

**Review by recuero — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**